src/api/Versions.js

```diff
@@ -242,6 +190,54 @@ class Versions extends OffsetBasedAPI {
         this.offsetGet(fileId, successCallback, errorCallback, offset, limit, fields, shouldFetchAll);
     }
 
+    /**
+     * API for fetching the current version for a file
+     *
+     * @param {string} fileId - a box file id
+     * @param {string} fileVersionId - a box file version id
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     * @returns {void}
+     */
+    getCurrentVersion(
+        fileId: string,
+        fileVersionId: string,
+        successCallback: BoxItemVersion => void,
+        errorCallback: ElementsErrorCallback,
+    ): void {
+        this.errorCode = ERROR_CODE_FETCH_CURRENT_VERSION;
+        this.get({
+            id: fileId,
+            successCallback,
+            errorCallback,
+            url: this.getVersionUrl(fileId, fileVersionId),
+            requestData: {
+                params: {
+                    fields: FILE_VERSIONS_FIELDS_TO_FETCH.toString(),
+                },
+            },
+        });
+    }
+
+    /**
+     * Decorates the current version as a FileVersions with restored information
+     *
+     * @param {BoxItemVersion} version - a box version
+     * @param {BoxItem} file - a box file
+     * @returns {FileVersions} - a FileVersions object containing the decorated current version
+     */
+    decorateCurrentVersion(version: BoxItemVersion, file: BoxItem): FileVersions {
+        const restoredFromId = getProp(file, 'restored_from.id');
+        const isRestoredVersion = !!restoredFromId;
+
+        if (isRestoredVersion) {
+            version.action = VERSION_RESTORE_ACTION;
+            version.version_restored = version.version_number;
```

Comment:
It looks like this is re-applying the version's number to a separate field. I think we need to look up `restored_from.id` value in the versions array and use that version number, instead. It's probably also worth adding an integration test for this behavior since it's a little fragile.